windows/scripts/extract_kernels.py

```diff
@@ -177,23 +150,147 @@ def seven_zip_extract(source, wildcard_includes, wildcard_excludes=None,
 
     args.append(source)
 
+    print("    %s - %s" % (" ".join(args), dest_dir))
+
     # Run 7-Zip
-    proc = subprocess.Popen(args, stdout=DEVNULL, stderr=subprocess.PIPE,
-                            cwd=dest)
+    proc = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, cwd=dest_dir)
     _, stderr = proc.communicate()
 
-    return proc.returncode, stderr
+    if proc.returncode:
+        raise Exception(u'Failed to extract file %s from %s: "%s"' % (wildcard_includes, source, stderr))
+
+def seven_zip_list(source):
+    args = ['7z', 'l', source]
+    proc = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    stdout, stderr = proc.communicate()
+
+    files = []
 
+    for line in stdout.splitlines():
+        els = line.split()
+        if len(els) == 6:
```

Comment:
Can you maybe add a small comment just to explain the significance of these magic numbers? That way if the 7zip output format ever changes it's easier to update the script. Is `els[5]` a file name? A file path? It's not immediately clear what this function returns